front/components/agent_builder/settings/AccessSection.tsx

```diff
@@ -43,9 +44,11 @@ export function AccessSection() {
     return scope.value === "visible" ? EyeIcon : EyeSlashIcon;
   };
 
-  const slackDataSource = supportedDataSourceViews.find(
-    (dsv) => dsv.dataSource.connectorProvider === slackProvider
-  )?.dataSource;
+  const slackDataSource = slackProvider
+    ? supportedDataSourceViews.find(
+        (dsv) => dsv.dataSource.connectorProvider === slackProvider
+      )?.dataSource
+    : null;
```

Comment:
Is the rest of the code able to handle a null DS, though? e.g. in https://github.com/dust-tt/dust/blob/main/front/lib/swr/connectors.ts#L32